src/Stryker.Core/Stryker.Core/DashboardCompare/GitInfoProvider.cs

```diff
@@ -5,115 +5,123 @@
             {
                 return;
             }
 
-            if (repository != null)
-            {
-                Repository = repository;
-            }
-            else
-            {
-                Repository = CreateRepository();
-            }
+            this.Repository = repository ?? this.CreateRepository();
         }
 
         public string GetCurrentBranchName()
         {
-            if (Repository?.Branches != null)
-            {
-                foreach (var branch in Repository.Branches)
-                {
-                    if (branch.IsCurrentRepositoryHead)
-                    {
-                        return branch.FriendlyName;
-                    }
-                }
-            }
-
-            return string.Empty;
+            return this.Repository?.Branches?.FirstOrDefault(x => x.IsCurrentRepositoryHead)?.FriendlyName ?? string.Empty;
         }
 
         private IRepository CreateRepository()
         {
-            if (string.IsNullOrEmpty(RepositoryPath))
+            if (string.IsNullOrEmpty(this.RepositoryPath))
             {
                 throw new StrykerInputException("Could not locate git repository. Unable to determine git diff to filter mutants. Did you run inside a git repo? If not please disable the --diff feature.");
             }
-            else
-            {
-                return new Repository(RepositoryPath);
-            }
+
+            return new Repository(this.RepositoryPath);
         }
 
-        private void Checkout()
+        private void Checkout(GitSourceKinds gitSourceKind)
         {
             try
             {
-                var branch = Repository.CreateBranch(_options.GitSource, $"origin/{_options.GitSource}");
-
-                Commands.Checkout(Repository, branch);
+                var branchName = gitSourceKind == GitSourceKinds.FriendlyBranchName ? this._options.GitSource : this.GetFriendlyName(this._options.GitSource);
 
-                var currentBranch = Repository.CreateBranch(_options.ProjectVersion, $"origin/{_options.ProjectVersion}");
+                this._logger.LogDebug($"Creating branch ${branchName} with committish origin/{branchName}");
+                var branch = this.Repository.CreateBranch(this._options.ProjectVersion, $"origin/{this._options.ProjectVersion}");
+                this._logger.LogDebug($"Checking out branch ${branchName}");
+                Commands.Checkout(this.Repository, branch);
 
-                Commands.Checkout(Repository, currentBranch);
+                this._logger.LogDebug($"Creating branch ${this._options.ProjectVersion} with committish origin/{this._options.ProjectVersion}");
+                var currentBranch = this.Repository.CreateBranch(this._options.ProjectVersion, $"origin/{this._options.ProjectVersion}");
+                this._logger.LogDebug($"Checking out branch ${this._options.ProjectVersion}");
+                Commands.Checkout(this.Repository, currentBranch);
             }
```

Comment:
@Mobrockers @Stijn-Rutten Could you confirm that behavior is correct ?

Reply:
This looks correct with the original code. However looking at it, this checks out the version given as the project version, however that does not have to be the original commit when stryker started. For example in a pull request scenario you would be on the wrong commit after this action. Perhaps we should cache the current head before doing the commit operation and checkout back to that commit instead of the projectversion. @Stijn-Rutten what do you think

Reply:
Hi from my point of view, in pr scenanio we should compare the whole merged branch (i.e refs/pulls/*/merge) to the `GitSource` branch (i.e refs/heads/develop). This is what azure devops is doing to display changes between 2 branches in the pr dashboard AFAIK.

Reply:
The correct commit should be the one that is checked out at the time stryker starts, no? So we should simply be able to request the current commit, save the id and checkout that commit again after the steps here.

I don't think another cmdline argument is the way to go here.

Reply:
In pr scenario, we may want to merge several commits and a merge commit from azure devops into `GitSource` branch.
@Mobrockers  do you think the current commit contains all the difference with the `GitSource` branch ?

Reply:
Also this implies using a way to detect that we are running in pr context.

Reply:
@Mobrockers i pushed an update

Reply:
I can't find a package associated with my pr could you please generate one ?

Reply:
@Mobrockers I implemented the suggested behavior. 
As is i can't really add unit tests to the Checkout method the implementation relies on the static Commands class.
Either we keep it as is or i can introduce adpaters to unit test the logic. Please let me know.
Also is there a way to get a pre version package from a feed i may add to a build to test wthe behavior within a pr in azure devops ?

Reply:
Hi sorry I did not get a notification about this comment! Yes there is a beta package feed here: https://www.myget.org/F/stryker-mutator-integrationtest

Unfortunately because myget only allows 100MB usage for open source projects a maximum of 3 pre-production versions van be available on the feed at a time, so your version will be gone soon after the pipeline has run.

The pre-production version number should correspond to the version number of the pipeline run for your pull request which you can check here: https://dev.azure.com/stryker-mutator/Stryker/_build?definitionId=4

You can check here if your package version is available: https://www.myget.org/feed/Packages/stryker-mutator-integrationtest